lib/widgets/app.dart

```diff
@@ -151,10 +153,13 @@ class ZulipApp extends StatefulWidget {
 }
 
 class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
+  late final Future<GlobalStore> _globalStoreFuture;
+
   @override
   void initState() {
     super.initState();
     WidgetsBinding.instance.addObserver(this);
+    _globalStoreFuture = ZulipBinding.instance.getGlobalStoreUniquely();
```

Comment:
This can do the job of both DeferredBuilderWidget and the Future.wait, in just a few lines of code because specialized to our use case:
```suggestion
    () async {
      final globalStore = await ZulipBinding.instance.getGlobalStoreUniquely();
      final notifFuture = NotificationOpenManager.instance.initializationFuture;
      if (notifFuture != null) await notifFuture;
      if (!mounted) return;
      setState(() {
        this.globalStore = globalStore;
      });
    }();
```

Relative to the status quo in main (or to a version after a refactor like the one in your first commit, moving the loading up to ZulipApp), this differs only in adding the two lines for `notifFuture`.

Because the underlying asynchronous work behind `initializationFuture` is already started by `main` before reaching this point, the only thing this needs to do to make the loading happen concurrently is to be sure to call `getGlobalStoreUniquely()` before awaiting `initializationFuture`.